hand-eye-calibration/full_hand_eye_sample.py

```diff
@@ -0,0 +1,486 @@
+
+    returns:
+        save_dir: Location where dataset is saved
+
+    Universal Robot registers:
+        output_int_register_24 = image_count:
+            Counter from robot side of number of images and poses taken.
+        output_bit_register_64 = start_capture:
+            Bool that trigger camera to capture an image
+
+        Signals to UR robot:
+        input_bit_register_64 = finish_capture:
+            Bool that trigger UR robot to move to next position.
+        input_bit_register_65 = ready_to_capture:
+            Bool that tells UR robot that the camera is ready to be used.
+    """
+
+    with zivid.Application() as app:
+        with app.connect_camera() as cam:
+
+            _set_camera_settings(cam)
+            ready_to_capture = True
+
+            save_dir = _generate_folder()
+
+            # Signal robot that camera is ready to capture
+            _write_robot_state(
+                con, input_data, finish_capture=False, camera_ready=ready_to_capture
+            )
+            robot_state = con.receive()
+
+            print(
+                "Initial output robot_states: \n"
+                f"Image count: {robot_state.output_int_register_24} \n"
+                f"Start capture: {robot_state.output_bit_register_64}"
+            )
+
+            images_captured = 1
+            while robot_state.output_int_register_24 != -1:
+
+                robot_state = con.receive()
+
+                if robot_state is None:
+                    print("Not able to recieve robot_state")
+                    con.disconnect()
+                    break
+
+                if (
+                    robot_state.output_bit_register_64
+                    and images_captured == robot_state.output_int_register_24
+                ):
+                    images_captured += 1
+
+                    print(
+                        f"[CAMERA] Capture image {robot_state.output_int_register_24}"
+                    )
+                    (
+                        frame,
+                        image_num,
+                        robot_pose_before,
+                        robot_pose_after,
+                    ) = _capture_and_get_robot_pose(con, cam)
+                    transform = _calculate_transform_matrix(
+                        robot_pose_before, robot_pose_after
+                    )
+
+                    # Signal robot to move to next position, then set to low again.
+                    _write_robot_state(
+                        con,
+                        input_data,
+                        finish_capture=True,
+                        camera_ready=ready_to_capture,
+                    )
+                    time.sleep(0.1)
+                    _write_robot_state(
+                        con,
+                        input_data,
+                        finish_capture=False,
+                        camera_ready=ready_to_capture,
+                    )
+
+                    # Using patlib to save images
```

Comment:
Typo. Also, not sure if this comment is necessary at all :)

Reply:
removed